docs/DeveloperGuide.md

```diff
@@ -189,104 +189,86 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+ * `Person#unmark()` — Removes the input time period from the person.
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If a command fails its execution, it will not call `Model#commitAddressBook()`, so the address book state will not be saved into the `addressBookStateList`.
+These operations make use of `Period#union()` and `Period#complement` and are exposed
+by `MarkCommand#execute()` and `RemoveMarkCommand#execute()`.
 
-</div>
 
-Step 4. The user now decides that adding the person was a mistake, and decides to undo that action by executing
-the `undo` command. The `undo` command will call `Model#undoAddressBook()`, which will shift the `currentStatePointer`
-once to the left, pointing it to the previous address book state, and restores the address book to that state.
+Given below is an example usage scenario and how the mark/unmark command
+changes the information stored.
 
-![UndoRedoState3](images/UndoRedoState3.png)
+Step 1. The user launches the application for the first time. The staff all
+the staff information is read from the save file `addressbook.json`.
+The initial information of the staff that we are looking at 
+will look something like this. The `Period` object that we are using can be taken to
+represent 1/1/2001 to 2/1/2001
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `currentStatePointer` is at index 0, pointing to the initial AddressBook state, then there are no previous AddressBook states to restore. The `undo` command uses `Model#canUndoAddressBook()` to check if this is the case. If so, it will return an error to the user rather
-than attempting to perform the undo.
 
-</div>
+![state0](images/MarkState0.png)
 
-The following sequence diagram shows how the undo operation works:
 
-![UndoSequenceDiagram](images/UndoSequenceDiagram.png)
+Step 2. The user executes the command `mark -i 1 d/2001-01-04`. The `mark` command
+calls `ParserUtil#parsePeriod()`, to obtain the period to mark, and `Person#mark()` to mark this period on this person. This method uses
+`Period#union()` to union the period with the set.
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
 
-</div>
+![state1](images/MarkState1.png)
 
-The `redo` command does the opposite — it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once
-to the right, pointing to the previously undone state, and restores the address book to that state.
+Step 3. The user executes the command `mark -i 1 d/2001-01-03`. The `mark` command
+calls `ParserUtil#parsePeriod()`, to obtain the period to mark as before. As this period makes the current range, 1/1/2001, 2/1/2001, 4/1/2001
+become contiguous, `Period#union()` that is called, merges the three `Period`
```

Comment:
Not sure if you wanna find a synonym for contiguous too it's a bit of a rare term but it's uty!